pkg/apis/openebs.io/scp/v1alpha1/storage-volume.go

```diff
@@ -39,3 +44,266 @@ type StorageVolumeList struct {
 
 	Items []StorageVolume `json:"items"`
 }
+
+// StorageVolumeSpec defines the desired spec of StorageVolume
+// +k8s:openapi-gen=true
+type StorageVolumeSpec struct {
+	// Capacity of the volume
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Capacity resource.Quantity `json:"capacity"`
+
+	// ProtocolConfiguration represents protocol specific configurations for NVME
+	// +nullable
+	ProtocolConfiguration ProtocolConfiguration `json:"protocolConfiguration,omitempty"`
+
+	// VolumeType stores configuration that is passed through storage class
+	VolumeType VolumeType `json:"volumeType"`
+
+	// Affinity is a group of affinity scheduling rules
+	// +nullable
+	Affinity Affinity `json:"affinity,omitempty"`
+
+	// StorageCohortID represents the cohort in which volume has to be provisioned
+	StorageCohortName string `json:"storageCohortName"`
+
+	// StoragePoolID represents the pool in which volume has to be created
+	StoragePoolName string `json:"storagePoolName"`
```

Comment:
Id or name?

Reply:
I had given a comment on whether we can use `name` instead of `id`
Benefits are -
1. There is no get by id in k8s client if we want to get it we need to list and filter by id
2. If we use name in case of delete and recreate it will work without any edit, For id we need to change the id for delete and recreate use case.

Reply:
I am ok with both though.

Reply:
What's the standard practice in K8s objects? We can use names if there are no possibilities of duplicate objects. eg. same pool name in same or different cohorts.

Reply:
> What's the standard practice in K8s objects?

In k8s it is name or full objectmeta and typemeta.
name is used in pvc,pc storage class is referenced by the name.
full objectmenta nad typemeta is used in the owner reference

Reply:
> We can use names if there are no possibilities of duplicate objects. eg. same pool name in same or different cohorts

How do we want to name our pool cr if there are pools with the same name in different cohorts?

Reply:
Depends on if we need to mandate name uniqueness or not. IMO, we should support same named entities across cohorts. Need to be discussed.